src/main/java/jimmy/mcgymmy/model/food/Carbohydrate.java

```diff
@@ -1,22 +1,39 @@
 package jimmy.mcgymmy.model.food;
 
+import jimmy.mcgymmy.commons.exceptions.IllegalValueException;
+
 /**
  * Represents Carbodydrates in food item in McGymmy.
  */
 public class Carbohydrate extends Macronutrient {
     public static final String MESSAGE_CONSTRAINTS =
             "Carbohydrate " + Macronutrient.MESSAGE_CONSTRAINTS;
     private static final int CARBOHYDRATE_MULTIPLIER = 4;
+    private static final int DEFAULT_CARBOHYDRATE = 0;
 
     /**
      * @param amount A valid amount.
      */
-    public Carbohydrate(int amount) {
+    public Carbohydrate(int amount) throws IllegalValueException {
         super(amount, CARBOHYDRATE_MULTIPLIER);
     }
 
-    public Carbohydrate(String amount) {
+    public Carbohydrate(String amount) throws IllegalValueException {
         this(Integer.parseInt(amount));
     }
 
+    /**
+     * Create a new Carbohydrate using the default value.
+     * Mostly boilerplate code.
+     * @return new Carbohydrate with default value.
+     */
+    public static Carbohydrate newDefault() {
+        try {
+            return new Carbohydrate(DEFAULT_CARBOHYDRATE);
+        } catch (IllegalValueException e) {
+            assert false : "Error in creation of default";
+            throw new IllegalArgumentException("Error in creation using default values.");
+        }
```

Comment:
Is this try-catch necessary?

Reply:
Yeah since Macronutrient's constructor throws IVE we need to catch it somewhere when creating the subclasses

actually just thought of a better alternative, will implement it soon

Reply:
EDIT nvm, Java is far too inflexible. This is probably the best solution we have.